### PR TITLE
auth: Do not send UID with external auth

### PR DESCRIPTION
**Pull request by idleroamer**
Due to mismatch between UID in a user-namespace and out-of-band credential acquired by server
on another user-namespace refrain from sending UID with authentication message

https://github.com/godbus/dbus/issues/345

### CODE REVIEW

**Comment by guelfey** [comment]
Thanks. I agree that this is probably better as a default behavior, but wouldn't want to break the API for this (although probably there are not many users that use AuthExternal directly). Could you add this as a separate `Auth` implementation (which is then used in `Conn.Auth` as the new default first method)?

**Review by guelfey — COMMENTED** [review]
also please check out the linter error

**Comment by idleroamer** [comment]
Somehow I tested with older godbus version, this already fixed in master as part of
https://github.com/godbus/dbus/issues/264

nevertheless I added a test-case, can't hurt

**Comment by guelfey** [comment]
Thanks - could you scope the test to only be compiled on Linux? It doesn't work on FreeBSD for obvious reasons and thus would break the build there

**Comment by idleroamer** [comment]
@guelfey Thanks for the review, test is conditionalized

File: conn_linux_test.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
+//go:build !freebsd
```
Comment (guelfey) [inline conn_linux_test.go:1]: this is a using a Linux-only syscall - we also have other non-Linux platforms besides FreeBSD that are supported (e.g. Darwin), please adjust accordingly

Comment (idleroamer) [reply conn_linux_test.go]: I see, adjusted accordingly, please recheck.

Comment (guelfey) [reply conn_linux_test.go]: you can just add `linux` as a build tag 😄 but I only now just noticed that you already put this in a file with a `_linux` suffix, which implicitly adds the corresponding build tag, so just removing the comments should be fine

Comment (idleroamer) [reply conn_linux_test.go]: thanks for the input, speaking using sledgehammer to crack a nut :)

**Comment by guelfey** [comment]
Thanks!

**Pull request merged**